src/lib/slider/slider.ts

```diff
@@ -226,6 +226,12 @@ export class MdSlider implements ControlValueAccessor {
   set vertical(value: any) { this._vertical = coerceBooleanProperty(value); }
   private _vertical = false;
 
+  /** The value to be used for display purposes. */
+  get displayValue(): string|number {
+    // Skip adding the decimal part if the number is whole.
+    return this.value % 1 === 0 ? this.value : this.value.toFixed(1);
```

Comment:
Using one decimal place is too prescriptive, since having a step of, say `0.05` between `0` and `1` is valid. It would be nice to have the slider intelligently figure out the appropriate number of decimal places to show based on the step value.